document-store/src/main/java/org/hypertrace/core/documentstore/mongo/query/parser/MongoFilterTypeExpressionParser.java

```diff
@@ -32,7 +33,9 @@ public Map<String, Object> visit(final RelationalExpression expression) {
   @SuppressWarnings("unchecked")
   @Override
   public Map<String, Object> visit(final KeyExpression expression) {
-    return Map.of(ID_KEY, expression.getKey().toString());
+    return Map.of(
+        ID_KEY,
+        Map.of("$in", expression.getKeys().stream().map(Key::toString).toArray(String[]::new)));
```

Comment:
I think we will have to check the performance of `in` and `eq` for a single key.

Reply:
Reverted to use `eq` for a single key.